numpy/linalg/umath_linalg.c.src

```diff
@@ -882,20 +787,28 @@ zero_@TYPE@_matrix(void *dst_in, const LINEARIZE_DATA_t* data)
                /* identity square matrix generation */
 /**begin repeat
    #TYPE = FLOAT, DOUBLE, CFLOAT, CDOUBLE#
-   #typ = float, double, COMPLEX_t, DOUBLECOMPLEX_t#
+   #typ = float, double, npy_cfloat, npy_cdouble#
    #cblas_type = s, d, c, z#
+   #suffix = f,  , f,  #
+   #cmplx = 0, 0, 1, 1#
  */
 static NPY_INLINE void
 identity_@TYPE@_matrix(void *ptr, size_t n)
 {
     size_t i;
     @typ@ *matrix = (@typ@*) ptr;
+#if @cmplx@
+    @typ@ one = npy_cpack@suffix@(1.0@suffix@, 0.0@suffix@);
+#else
+    @typ@ one = 1.0@suffix@;
+#endif
```

Comment:
IMO this type of change makes things harder not easier to read.

Reply:
I can revert this kind of change.  Probably not until sometime tomorrow.

Reply:
I've modified this to be more like the original.